### PR TITLE
Fix bug in flaky test script

### LINKED ISSUE #15800: Flaky test is flaky
### description

there is a test to report on apparent test flakes - but that itself is throwing an error:

https://dev.azure.com/cncf/envoy/_build/results?buildId=70906&view=logs&j=8c169225-0ae8-53bd-947f-07cb81846cb5&t=6b0ace90-28b2-51d9-2951-b1fd35e67dec&l=23349

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

Commit Message:

Fixes #15800

There are some errors such as `Error: No such container: rbe-container-453b391c-e65b-4035-bfa6-fee0cdb0cd00` where the split fails. Harden the accesses with length checks.

Additional Description:
Risk Level: Low
Testing: Manual
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
Verifying on the CI and will assign a reviewer

File: ci/flaky_test/process_xml.py [hunk 1]
```diff
@@ -103,8 +103,17 @@ def parse_and_print_test_suite_error(testsuite, log_path):
             # finding the last test case to run. The expected format of that is:
             #     "[ RUN      ] <TestParams>/<TestSuite>.<TestCase>\n".
             last_test_fullname = test_output.split('[ RUN      ]')[-1].splitlines()[0]
-            last_testsuite = last_test_fullname.split('/')[1].split('.')[0]
-            last_testcase = last_test_fullname.split('.')[1]
+            last_test_fullname_splitted_on_slash = last_test_fullname.split('/')
```
Comment (antoniovicente) [inline ci/flaky_test/process_xml.py:106]: IIRC the `<TestParams>/` portion of the run name is optional.

I think you should split by '.' first and then optionally split on / if present.  Something like:
```
            last_test_fullname_splitted_on_dot = last_test_fullname.split('.')
            if len(last_test_fullname_splitted_on_dot) == 2:
                last_testcase = last_test_fullname_splitted_on_dot[1]
                testsuite_split_on_slash = 
                last_testsuite = last_test_fullname_splitted_on_dot[0].split('/')[-1]
            else:
                last_testcase = "Could not retrieve last test case"
                last_testsuite = "Could not retrieve last test suite"
```

Comment (davinci26) [reply ci/flaky_test/process_xml.py:106]: Is this format enforced? I thought we were parsing a string so I was trying to be as defensive as possible with my assumptions

File: ci/windows_ci_steps.sh [hunk 2]
```diff
@@ -42,7 +42,7 @@ fi
 # Environment setup.
 export TEST_TMPDIR=${BUILD_DIR}/tmp
 
-[[ "${BUILD_REASON}" != "PullRequest" ]] && BAZEL_EXTRA_TEST_OPTIONS+=(--nocache_test_results)
+#[[ "${BUILD_REASON}" != "PullRequest" ]] && BAZEL_EXTRA_TEST_OPTIONS+=(--nocache_test_results)
```
[the remark this answers is not in the document]
Comment (davinci26) [reply ci/windows_ci_steps.sh:45]: Verified in the slack channel that it works

File: ci/flaky_test/process_xml.py (hunk 1 above)
Comment (antoniovicente) [reply ci/flaky_test/process_xml.py]: I was just pointing out that your version didn't correctly extract test names if the test is not parametrized.  I do appreciate being as defensive as possible.

Thanks for fixing this name extraction logic.

**Review by antoniovicente — APPROVED** [review]
Thanks for the fix.

**Pull request merged**